lib/cmorizer.rb

```diff
@@ -15,10 +15,30 @@ def initialize(src_txt)
       @experiments = []
       @years_step = 1
       @eval_mode = true
+      src_txt = sanitize src_txt
       instance_eval(src_txt, src_txt)
       @eval_mode = false
     end
     
+    def sanitize(src_txt)
+  # Sanitizes the source text to avoid mistreating variable names as Ruby constants
+      new_txt = []
+      requires_sanitization = false
+      src_txt.each_line do |line|
+        if line.start_with?("cmorize ") && line.include?("=>")
+          first, varname, *rest = line.split
+          if varname[0] =~ /[A-Z]/
+            requires_sanitization = true
+            new_txt << "#{varname} = '#{varname}'"
+          end
+        end
+      new_txt << line
+      end
+      if requires_sanitization
+        src_txt = new_txt.join("\n")
+      end
```

Comment:
Shouldn't this be part of the `src_txt.each_line do |line|` loop? I really don't know RUby enough to understand whether this is correct or not.

Reply:
This should be outside the loop. Inside the loop `new_txt` Array gets populated with lines irrespective of `requires_sanitization`. Outside the loop,  this `if requires_sanitization` condition tells whether original text modification is required. In principal, this check is not really required but treading carefully not to use `new_txt` if it not required. Sort of avoiding context for unnecessary bugs.

Reply:
I see, thanks for the clarification.